### PR TITLE
#3q303pf Reworked Wallet UI

### PR DESCRIPTION
**Pull request by saadahmsiddiqui**
## Issue
[#3q303pf Reworked Wallet UI](https://app.clickup.com/t/3q303pf)

### CODE REVIEW

**Comment by ComposableFiB0t** [comment]
Task linked: [CU-3q303pf Reworked Wallet UI](https://app.clickup.com/t/3q303pf)

File: frontend/packages/polkadot-wallet/components/ConnectWalletModal.tsx [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+                    width: "100%",
+                    display: "flex",
+                    flexDirection: "column",
+                    gap: 4,
+
+                }}
+            >
+                {accounts.map((account) => (
+                    <PolkadotAccount
+                        account={account}
```
Comment (easteregg) [inline frontend/packages/polkadot-wallet/components/ConnectWalletModal.tsx:121]: `key` prop is necessary here.

File: frontend/packages/polkadot-wallet/components/ConnectWalletModal.tsx [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+                        onSelect={setSelectedActiveAccount}
+                        isSelected={selectedActiveAccount ? selectedActiveAccount.address === account.address : false}
```
Comment (easteregg) [inline frontend/packages/polkadot-wallet/components/ConnectWalletModal.tsx:123]:
```suggestion
                        isSelected={!!selectedActiveAccount?.address || false}
```

File: frontend/packages/polkadot-wallet/components/ConnectWalletModal.tsx [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+        return supportedPolkadotWallets.map(wallet => (
+            <ConnectListItem
+                onClick={(walletId: SupportedWalletId) => {
+                    onConnectPolkadotWallet(walletId).then((walletConnected) => {
+                        setWalletConnectStep(WalletConnectStep.SelectDotsamaAccount);
+                    });
+                }}
+                name={wallet.name}
+                icon={wallet.icon}
+                id={wallet.walletId}
```
Comment (easteregg) [inline frontend/packages/polkadot-wallet/components/ConnectWalletModal.tsx:199]: `key` prop is necessary

File: frontend/packages/polkadot-wallet/components/ConnectWalletModal.tsx [hunk 4]
```diff
@@ -0,0 +1,310 @@
...
+        switch (walletConnectStep) {
+            case WalletConnectStep.SelectDotsamaAccount:
+                setWalletConnectStep(WalletConnectStep.SelectedDotsamaWallet);
+                break;
+            case WalletConnectStep.SelectedDotsamaWallet:
+            case WalletConnectStep.SelectEthereumWallet:
+                setWalletConnectStep(WalletConnectStep.SelectNetwork);
+                break;
+        }
+    }, [walletConnectStep]);
```
Comment (easteregg) [inline frontend/packages/polkadot-wallet/components/ConnectWalletModal.tsx:251]: The computation is not heavy to rely on useMemo, all of these can be moved to their own pure functions and to be used here.

**Review by easteregg — APPROVED** [review]

**Review by andresmechali — APPROVED** [review]

**Pull request merged**